### PR TITLE
Create strategies-for-upgrading-grafana

### PR DESCRIPTION
**Pull request by irenerl24**
This PR creates a new page with the following content:
- How to upgrade Grafana
- Types of releases
- Dates of each release

**What is this feature?**

This page is created from a blog post. As per @margotphelps request, this will be created as a permanent page in our docs site so that the release dates can be edited. 

**Why do we need this feature?**

Having this page in our doc site instead of a blog post allows us to make changes easily. 

**Who is this feature for?**

All users of Grafana on-prem

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by jdbaldry — COMMENTED** [review]
Comments that apply to the whole page:

- Its not a bad idea to replace all the smart quotes in this with ASCII alternatives: https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanasmartquotes
- This does read a lot more like a blog than documentation
- Might be worth finding a more descriptive term than "ad-hoc" where it's used through the page

**Review by jdbaldry — APPROVED** [review]
Still feels very blog-y to me but I think having the documentation is of value in itself and I suppose we can rewrite it later.

I think Isabel should take a look as a CODEOWNER before this is merged.

**Comment by irenerl24** [comment]
Thank you for all your feedback as usal, Jack! Definitely I'll wait for Isabel for review of the doc :)

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:27]:
```suggestion
- **Minor release (every other month)**: These releases can include new features, deprecation notices, notices about upcoming breaking changes, previously announced breaking changes, bug fixes, and security vulnerability patches.
- **Major release (once a year, in April/May)**: These are like a minor release, but accompanied by [GrafanaCON](https://grafana.com/events/grafanacon/) and a comprehensive upgrade guide for users who like to upgrade only once a year.
- **Patching release (every month)**: These include bug fixes for currently supported versions, as well as any security vulnerability patches.
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:34]:
```suggestion
Whether it’s knowing that a bug has been fixed, seeing that a security vulnerability is patched, or understanding how to mitigate the impact of breaking changes, proper documentation allows you to make informed decisions about when to upgrade your local Grafana instances.
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:45]: Let's make this evergreen
```suggestion
Currently, Grafana is on a monthly release cycle, with release types rotating every other month between minor or patching. For example, here’s a look at scheduled releases for 2024:
```

Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:45]: This description doesn't align with what's in the table or the **What to expect from each release type** section. I'm not seeing a rotation of any kind. We do patching monthly, minor every other month:

> with release types rotating every other month between minor or patching.

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:59]: Making this more general vs specifically about our 2024 release dates. 
```suggestion
- The schedule above outlines how we plan release dates. However, unforeseen events and circumstances may cause dates to change.
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:91]:
```suggestion
- The current (most recently released) version of Grafana gets the highest level of support. Releases for this version include all the new features along with all bug fixes.
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:102]: We should find another word here
```suggestion
- Major (critical) impact to a significant amount of customers.
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:111]:
```suggestion
| **Minor / bi-monthly (11.1 to 11.2)**     | Our recommended strategy. It combines up-to-date, secure releases with access to latest features as soon as they're released. <ul><li>Small changelog to review</li><li>Highest compatibility with actively maintained plugins</li><li>Easy migration to [Grafana Cloud](https://grafana.com/products/cloud)</li></ul> | <ul><li>**June 2024**: You review the 11.1 changelog and deploy the release to testing</li><li>**July 2024**: You deploy 11.1 to production</li><li>**August 2024**: 11.2 is released</li></ul>                                                                             |
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:112]:
```suggestion
| **Major / yearly (10.0 to 11.0)**         | Yearly upgrade path that still gives access to up-to-date features presented at GrafanaCON.<ul><li>Big changelog to review</li><li>High compatibility with plugins</li><li>Relatively easy migration to [Grafana Cloud](https://grafana.com/products/cloud)</li></ul>                                                   | <ul><li>**April 2024**: 11.0 is released, you start a big changelog review</li><li>**May 2024**: You deploy 11.0 to testing</li><li>**June 2024**: You deploy 11.0 to production</li><li>**April 2025**: 12.0 is released</li></ul>                                                 |
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:115]:
```suggestion
For each strategy, you should stay informed about patch releases that fix security vulnerabilities (released monthly, plus ad-hoc releases). Follow the “minor” strategy for the most flexibility, as you can also occasionally lengthen the cadence to a full quarter and still rely on your currently deployed minor release being supported with security fixes.
```

**Review by imatwawana — COMMENTED** [review]
I've mostly made wording fixes to remove idiomatic phrases and a few other formatting things. There's one question I have where something doesn't make sense to me.

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (imatwawana) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:45]: This part still doesn't make sense to me:

> Currently, Grafana is on a monthly release cycle, with release types rotating every other month between minor or patching.

Based on the calendar below, I'm not clear on how this is rotating every other month when patching releases are every month.

**Review by imatwawana — COMMENTED** [review]
Just one unresolved comment from before. The first sentence on line 45 doesn't seem to be accurate.

File: docs/sources/upgrade-guide/when-to-upgrade/index.md
Comment (irenerl24) [reply docs/sources/upgrade-guide/when-to-upgrade/index.md:45]: that's one for @margotphelps to clarify :)

Comment (margotphelps) [reply docs/sources/upgrade-guide/when-to-upgrade/index.md:45]: You are correct, originally the schedule was every other month for patching but we changed that you can remove ", with release types rotating every other month between minor or patching. For example,"

**Review by margotphelps — CHANGES_REQUESTED** [review]
There is one clean up item that you noted about the release rotating between patching and Minor that needs to be addressed and then this can be merged. I will be making some more revisions probably next week once we have the November and December releases solidified.

**Review by margotphelps — APPROVED** [review]

**Comment by irenerl24** [comment]
Thank you all for your collaboration in this PR :) Much appreciated!

**Pull request merged**
